org_status/org_hosts/gitlab.py

```diff
@@ -15,31 +15,32 @@ class GitLabOrg(OrgHost):
     HOST_STATUS_URL = ('https://api.status.io/1.0/status'
                        '/5b36dc6502d06804c08349f7')
 
-    def __init__(self, token, group, **kargs):
+    def __init__(self, token, group, fetch, **kargs):
         super().__init__(**kargs)
 
         self._group = group
-        self._token = GitLabPrivateToken(token)
-        self._org = GitLabOrganization(self._token, self._group)
-
         self._status_provider = self.StatusProvider(self._group)
 
+        if fetch:
+            self._token = GitLabPrivateToken(token)
+            self._org = GitLabOrganization(self._token, self._group)
+
     @classmethod
     def get_host_status(cls):
         status = requests.get(cls.HOST_STATUS_URL)
         status = json.loads(status.text)
         return status['result']['status_overall']['status'] == 'Operational'
 
-    def process_repository(self, repo, branch='master'):
-        self.print_status(repo.web_url)
+    def process_repository(self, web_url, branch='master'):
+        self.print_status(web_url)
 
         # reliable enough?
-        repo_name = '/'.join(repo.web_url.split('/')[4:])
+        repo_name = '/'.join(web_url.split('/')[4:])
         repo_status = self._status_provider.get_status(repo_name,
                                                        self.HostName,
                                                        branch=branch)
 
-        return RepoStatus(repo.web_url, repo_status)
+        return RepoStatus(web_url, repo_status)
 
     @property
     def repositories(self):
```

Comment:
You should also fix this property to return `web_url's` rather than `Repository` objects in both the `sync` states, this way we have predictable and standard behavior.